client/services/code_evaluators/CodeRunnerDispatcherServiceSpec.js

```diff
@@ -31,6 +31,37 @@ describe('CodeRunnerDispatcherService', function() {
       };
       expect(errorFunction).toThrowError(Error);
     });
+
+    it('should throw a timeout error if passed infinitely looping code',
+      function() {
+        var code =
+        `while True:
+          print 'a'
+        `;
+
+        CodeRunnerDispatcherService.runCodeAsync("python", code).then(
+          function(rawCodeEvalResult) {
+            expect(rawCodeEvalResult.getErrorString()).toBe(
+              'TimeLimitError: Program exceeded run time limit. on line 2')
+          });
+    });
+
+    it('should throw a timeout error if passed infinitely recursing code',
```

Comment:
The error message here looks more like "exceeded call stack size" than timeout...